Framework/Algorithms/src/LorentzCorrection.cpp

```diff
@@ -108,8 +172,47 @@ void LorentzCorrection::exec() {
     PARALLEL_END_INTERUPT_REGION
   }
   PARALLEL_CHECK_INTERUPT_REGION
+}
+
+void LorentzCorrection::processTOF_PD(MatrixWorkspace_sptr &wksp,
+                                      Progress &prog) {
+  const int64_t numHistos = static_cast<int64_t>(wksp->getNumberHistograms());
+  const auto &spectrumInfo = wksp->spectrumInfo();
+
+  EventWorkspace_sptr wkspEvent =
+      boost::dynamic_pointer_cast<EventWorkspace>(wksp);
+  bool isEvent = bool(wkspEvent);
+
+  PARALLEL_FOR_IF(Kernel::threadSafe(*wksp))
+  for (int64_t i = 0; i < numHistos; ++i) {
+    PARALLEL_START_INTERUPT_REGION
 
-  this->setProperty("OutputWorkspace", outWS);
+    if (!spectrumInfo.hasDetectors(i))
+      continue;
+
+    const double sinTheta_v = sinTheta(spectrumInfo, i);
+
+    const auto points = wksp->points(i);
+    const auto pos = std::find(cbegin(points), cend(points), 0.0);
+    if (pos != cend(points)) {
+      std::stringstream buffer;
+      buffer << "Cannot have zero values Wavelength. At workspace index: "
+             << pos - cbegin(points);
+      throw std::runtime_error(buffer.str());
+    }
+
+    if (isEvent) {
+      wkspEvent->getSpectrum(i) *= sinTheta_v;
```

Comment:
No wavelength dependence???? I see that's what they asked, but are you sure they are right?